### PR TITLE
Make log dir optional

### PR DESCRIPTION
**Pull request by bcpeinhardt**
This makes passing the --log-dir option to coder/coder optional and based on whether a proxyLogDirectory is set in the extensions settings.

The reasoning behind this is rather than make a unilateral decision about where and how long to store the logs, we simply give control to the user.

### CODE REVIEW

File: src/remote.ts
Comment (code-asher) [inline src/remote.ts:553]:
```suggestion
```

There are no checks performed on the extension's version, it just depends on whether the setting exists in the `package.json` that is in that version.  If it were somehow missing I think it would just return undefined but I am not sure.  But someone would have to manually edit the `package.json` for that to happen.  :stuck_out_tongue:

File: CHANGELOG.md
Comment (code-asher) [inline CHANGELOG.md]: It could be helpful to add some description on why one might want to use the flag, and I think `--log-dir` could be considered an internal implementation detail so maybe we do not need to mention it.  Maybe something like:

```suggestion
- Previously, the extension would always log SSH proxy diagnostics to a fixed
  directory. Now this must be explicitly enabled by configuring a new setting
  `coder.proxyLogDirectory`. If you are having connectivity issues, please
  configure this setting and gather the logs before submitting an issue.
```

File: README.md
Comment (code-asher) [inline README.md]:
```suggestion
Use this when formatting paths in the Coder extension settings rather than ~ or
$HOME.
```

File: README.md
Comment (code-asher) [inline README.md]: I think the official spelling has a space:

```suggestion
${userHome} from VS Code's
```

File: src/remote.ts
Comment (code-asher) [inline src/remote.ts]:
```suggestion
    this.storage.writeToCoderOutputChannel(`SSH proxy diagnostics are being written to ${logDir}`)
```

File: src/util.ts
Comment (code-asher) [inline src/util.ts:66]: Very tiny nit/question: are these valuable with TypeScript?  My knee-jerk reaction is that at best it duplicates the info TypeScript gives and at worse it could get out of sync if someone changes the signature but not the comment.

But, maybe there is some IDE tooling that uses these.

**Review by code-asher — APPROVED** [review]
Looks great!  I only have some suggestions on wordings.  Feel free to merge whenever you feel it is ready.

File: src/util.ts
Comment (bcpeinhardt) [reply src/util.ts:66]: I don't have strong opinions on this at all, my default was just to document the function however typescript folks usually do. I'll leave it for now so you don't have to re-review haha.

Comment (code-asher) [reply src/util.ts:66]: Fair fair!  No worries about re-review for small things like this, by the way.  :)  For the most part once you have an approval you are free to merge.  If you make a big change or something though you might want a re-review.

**Pull request merged**
